content/blog/2016/2016-10-11-stage-lock-milestone.adoc

```diff
@@ -0,0 +1,165 @@
+
+=== Stage
+
+The `stage` step is a primary building block in Pipeline, dividing the steps of a Pipeline into explicit units and helping to visualize the progress. The visualization plugin is called "Stage View" due to this function of `stage`.
+
+Beginning with version 2.2 of `Pipeline Stage Step` plugin, the `stage` step now requires a block argument, wrapping all steps within the defined stage. This makes the boundaries of where each `stage` begins and ends obvious and predictable.
+
+In addition, the concurrency argument of `stage` has now been removed to make this step more concise; responsibility for concurrency control has been delegated to the `lock` step.
+
+[source, groovy]
+----
+stage("Build"){
+  doSomething()
+  sh 'echo $PATH'
+}
+----
+
+Omitting the block from `stage` and using the concurrency argument are now deprecated in Pipeline. Pipelines using this syntax will continue to function but will produce a warning in the console log:
+----
+Using the 'stage' step without a block argument is deprecated
+----
+This message is only a reminder to update your Pipeline scripts; none of your Pipelines will stop working. If we reach a point where the old syntax is to be removed we will make an announcement prior to the change. We do, however, recommend that you update your existing Pipelines to utilize the new syntax.
+
+*note:* Stage View and Blue Ocean will both work with either the old `stage` syntax or the new.
+
+=== Lock
+
+Rather than attempt to limit the number of concurrent builds of a job using the `stage`, we now rely on the `Lockable Resources` plugin and the `lock` step to control this. The `lock` step is limited to a concurrency of one (1) build but it provides much greater flexibility in designating where jobs should be restricted.
+
+* `lock` can be used to constrain just a subsection of a `stage`
+
+[source, groovy]
+----
+stage("Build"){
+  doSomething()
+    lock("myResource"){
+    echo "locked build"
+  }
+}
+----
+
+* `lock` can be used to wrap multiple stages into a single concurrency unit
+
+[source, groovy]
+----
+lock("myResource"){
+  stage("Build"){
+    echo "Building"
+  }
+  stage("Test"){
+    echo "Testing"
+  }
+}
+----
+
+=== Milestone
+
+The `milestone` step is the last piece of the puzzle to replace functionality originally intended for `stage` and adds even more control for handling concurrent builds of a job. The `lock` step limits the number of builds running concurrently in a section of your Pipeline while the `milestone` step ensures that older builds of a job will not overwrite or clobber a newer build.
+
+Concurrent builds of the same job do not always run at the same rate. Depending on the network, the node used, compilation times, test times, etc. it is always possible for a newer build to complete faster than an older build.
+
+* Build A is triggered
```

Comment:
There should be a "for example:" before this list.  e.g. after "... faster than an older build"